README.md

```diff
@@ -15,21 +15,135 @@ ready until after the talk.
-Firebase and _keep_ migrating new data. You can find this in the cloud
-folder.
+individual ParseObject would map to the Firebase Realtime
+Database. We offer two files to help you migrate from Parse:
+
+ 1. `migrator.js` A tool that allows you two keep your existing Cloud Code *and* use Cloud Code to migrate data out of Parse.
+ 2. `firebase-rest.js` A simple wrapper that helps you talk to Firebase in Cloud Code (where the Firebase Node module doesn't work).
+ 
+Imagine you have a `BlogPost` class in Parse with the following fields:
+
+* **title** *(String)* Name the post
+* **body** *(String)* Blog contets
+* **keywords** *(String)* Originally you chose to do text searches on a space separated list
+* **kewords_list** *(Array)* You realized later that Parse can search an array better and added Cloud Code to put **keywords** into **keywords_list**
+
+You probably have Cloud Code like:
+
+```js
+Parse.Cloud.beforeSave('BlogPost', function(request, response) {
+  var object = request.object;
+  if (object.isDirty('keywords_list')) {
+    response.error('keywords_list is managed by Cloud Code only');
+    return;
+  }
+  object.set('keywords_list', object.get('keywords').split(' '));
+  response.success(object);
+}
+```
+
+A logical structure for your blog posts in Firebase might look like:
+
+```json
+{
+  "posts": {
+    "aBlogPostID": {
+      "title": "Firebase expands to become a unified app platform",
+      "body": "Eighteen months ago, Firebase joined Google...",
+      "keywords": ["googleio", "announcements"],
+    },
+  },
+  "keywords": {
+    "googleio": {
+      "aBlogPostID": true,
+      "anotherBlogAboutGoogleIO": true
+    },
+    "announcements": {
+      "aBlogPostID": true
+    }
+  }
+}
+```
+
+A few things are noticeably different in this structure:
+
+1. We no longer use the legacy `"keywords"` array
+2. The values from `"keywords_list"` are stored in a different location so we can look up blogs related to a `"googleio"` by reading the keys under `"/keywords/googleio"`
+
+Migrator can help us do this very easily in 3 steps:
+
+1. Change our Parse.beforeSave to a Migrator.beforeSave. This is the same except it works with promises instead of callbacks.
+2. Describe how we would copy an object from Parse to Firebase
+3. (optional) Describe how we translate a Parse object deletion to Fireabse
+
+We would do this with the following Cloud Code:
+
+```js
+var _ = require('underscore');
+var migrator = require('cloud/migrator.js')(Parse);
+var ref = require('cloud/firebase.js')({
+  url: 'https://my-firebase-database-url.firebaseio.com',
+  key: 'if-my-firebase-database-is-secured-this-key-gives-me-admin-access',
+  parse: Parse
+});
+
+// Notice this is almost the same (and actually a bit simpler) than the original beforeSave
+migrator.beforeSave('BlogPost', function(request) {
+  var object = request.object;
+  if (object.isDirty('keywords_list')) {
+    throw 'keywords_list is managed by Cloud Code only';
```

Comment:
Are you sure this works as intended? I seem to recall jQuery promises (and therefore Parse.Promise) being weird about "throw".